### ISSUE TITLE
Hardware Carpentry

### ISSUE THREAD

**Issue by amandawhitmire** [carpentries-incubator/proposals#8]
I'm not sure if this falls under SWC or DC, but I've tweeted a bit about the idea of developing a "hardware carpentry" (HT: Jonah came up with the name). I think there is interest in this. I'll be developing a curriculum for this at my institution, and am more than happy to do this under the Carpentries umbrella. 

I am in the very early stages of thinking about this, but the focus is on ~Arduino microcontrollers, how they work, circuits, programming them, building a sensor, I/O, etc. Not sure if there is room, but maybe a module on ~Raspberry Pi stuff. I'll have more work on this done in the next couple of weeks.

**Comment by ErinBecker** [comment]
@anacost, @crdietrich, @amandawhitmire - We've just created The Carpentries Incubator GH organization to provide a single point of entry for community members to share lesson materials and discuss ideas for new lessons. Your lesson materials were previously listed in the lesson-ideas repo, which has now been deprecated. All of the issues in that repo have been migrated here. We're still working on organizing this repo, so please do let me know if you have ideas for how best to organize this information to make it accessible to the community.

**Comment by ErinBecker** [comment]
@anacost - any interest in hosting the UIO Hive lesson in the Incubator? Let me know if you're interested and I'd be happy to facilitate.

**Comment by annefou** [comment]
@ErinBecker That's a very good idea! Thanks @anacost for mentionning our lesson!

**Comment by ErinBecker** [comment]
@annefou - great! Happy to have this lesson in the Incubator. It looks like you're already using The Carpentries lesson template, so the only things required in order to include this lesson in the Incubator is a confirmation that the lesson a) has a CC-BY or CC-0 license and b) conforms to our Code of Conduct. Once those two requirements are fulfilled, I'll grant you access to transfer the repo to the Incubator. Let me know if you have any questions!

**Comment by annefou** [comment]
@ErinBecker Yes we are ready to move the UiO Hive lessons to the Incubator. We are all set with the license  https://uiohive.github.io/IoT_arduino_nano/LICENSE.html (CC BY 4.0 license as for Carpentries lessons) and follow the Carpentries Code of Conduct (https://uiohive.github.io/IoT_arduino_nano/CODE_OF_CONDUCT.html).

We will definitely participate to the community discussion around the incubator. Thanks!

**Comment by ErinBecker** [comment]
Great to hear @annefou! I've just sent you an invitation to join the carpentries-incubator organization. This will give you the ability to transfer your repos over. You'll need to go into each of the repos settings pages and click "transfer this repository" (in the danger zone). Once you've done that, I can accept the transfer and get things set up to have your lessons populate in the Incubator.

(Alternatively, you can give me ownership on your repo and I can transfer it for you, if you prefer that route.)

**Comment by limako** [comment]
Hi. I'm coming late to the party, but I'm developing a new class called "Open Science Instrumentation and Data Collection" where students will learn to use the Raspberry Pi and Arduino ecosystems to make scientific instruments (i.e. connect to sensors, collect & transform data, etc.). I'm tentatively planning to develop my instructional materials using the SWC template. I'm still putting together the full list of topics, but "hardware carpentry" sounds like a natural home for much of what I'm proposing to develop.

**Comment by fmichonneau** [comment]
@limako, ICYMI there is a lesson on using Arduino here: https://carpentries-incubator.github.io/IoT_arduino_nano/ (GitHub repo here: https://github.com/carpentries-incubator/IoT_arduino_nano/) in case it's useful and want to start contributing there!

**Comment by limako** [comment]
Thanks much! I'll check it out. With the start of the new semester, I'm working on my course development project full time now. I'm tentatively planning to have my students program their arduino devices via command line from a raspberry pi. I'm working up lessons on setting up the raspberry pi, using the GPIO, and then interfacing with and programming an arduino.

**Comment by limako** [comment]
I've posted the alpha version of my first lesson https://limako.github.io/rpi-arduino/ Feedback welcome!

I've also submitted it to the incubator proposals with the goal of it serving within a "hardware carpentry" universe of materials: https://github.com/carpentries-incubator/proposals/issues/36
